### PR TITLE
Match some ``frame.c`` functions

### PR DESCRIPTION
**Pull request by Yanis42**
couldn't figure out a better way than using a temp inside ``packFreeBlocks``

### CODE REVIEW

File: src/emulator/frame.c [hunk 1]
```diff
@@ -511,7 +691,29 @@ const f64 D_80135F88 = 8.44;
...
+    iPack = *piPack;
+
+    if (iPack == -1) {
+        return 1;
+    }
+
+    nMask = ((1 << ((iPack) >> 0x10)) - 1) << (iPack & 0x1F);
+    temp_r6 = (iPack & 0xFFFF) >> 5;
+
+    if (nMask == (u32)(nMask & anPack[temp_r6])) {
```
Comment (cadmic) [inline src/emulator/frame.c:709]: `(u32)` cast has no effect?

File: src/emulator/frame.c [hunk 2]
```diff
@@ -521,15 +723,32 @@ const f64 D_80135F88 = 8.44;
...
+    pFrame->aDraw[2] = (FrameDrawCallback)frameDrawRectFill_Setup;
+    pFrame->aDraw[3] = (FrameDrawCallback)frameDrawRectTexture_Setup;
+    return 1;
+}
+
+s32 frameSetFill(Frame* pFrame, s32 bFill) {
+    if (bFill) {
+        pFrame->nMode |= 0x20000;
+    } else {
+        pFrame->nMode &= 0xFFFDFFFF;
```
Comment (cadmic) [inline src/emulator/frame.c:739]: `~0x20000`

File: src/emulator/frame.c [hunk 3]
```diff
@@ -553,7 +775,13 @@ const f64 D_80135F88 = 8.44;
 
 #pragma GLOBAL_ASM("asm/non_matchings/frame/frameResetUCode.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/frame/frameSetBuffer.s")
+s32 frameSetBuffer(Frame* pFrame, FBTType eType) {
+    if (((u32)(eType - 2) > 1) && (eType == FBT_DEPTH)) {
```
Comment (cadmic) [inline src/emulator/frame.c:779]: hmm is this a switch statement?

Comment (Yanis42) [reply src/emulator/frame.c:779]: probably but I can't find anything that match

**Pull request merged**